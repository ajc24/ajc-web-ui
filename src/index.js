/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
