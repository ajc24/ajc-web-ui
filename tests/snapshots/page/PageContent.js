import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { PageContent } from '../../../src';

describe('PageContent', () => {
  describe('Default props and rendering - Component with white background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageContent title="Test Title">
            This is a test.
          </PageContent>
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with grey background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageContent title="Test Title" colour="grey">
            This is a test.
          </PageContent>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with transparent background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageContent title="Test Title" colour="transparent">
            This is a test.
          </PageContent>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with yellow background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageContent title="Test Title" colour="yellow">
            This is a test.
          </PageContent>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
