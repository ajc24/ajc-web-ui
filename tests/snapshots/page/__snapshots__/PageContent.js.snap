// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent Default props and rendering - Component with white background verifies the snapshot for the component 1`] = `
<main
  aria-label="Test Title"
  className="ajc-flex-row ajc-flex-justify-center ajc-page-content-panel ajc-spacing-top-small ajc-spacing-bottom-small ajc-width-4k ajc-background-white"
  id="main-content"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-width-hd"
    >
      <div
        className="ajc-flex-column ajc-width-hd-with-padding ajc-text-black ajc-font-default"
      >
        This is a test.
      </div>
    </div>
  </div>
</main>
`;

exports[`PageContent Transferred props and rendering - Component with grey background verifies the snapshot for the component 1`] = `
<main
  aria-label="Test Title"
  className="ajc-flex-row ajc-flex-justify-center ajc-page-content-panel ajc-spacing-top-small ajc-spacing-bottom-small ajc-width-4k ajc-background-grey-1"
  id="main-content"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-width-hd"
    >
      <div
        className="ajc-flex-column ajc-width-hd-with-padding ajc-text-black ajc-font-default"
      >
        This is a test.
      </div>
    </div>
  </div>
</main>
`;

exports[`PageContent Transferred props and rendering - Component with transparent background verifies the snapshot for the component 1`] = `
<main
  aria-label="Test Title"
  className="ajc-flex-row ajc-flex-justify-center ajc-page-content-panel ajc-spacing-top-small ajc-spacing-bottom-small ajc-width-4k ajc-background-transparent"
  id="main-content"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-width-hd"
    >
      <div
        className="ajc-flex-column ajc-width-hd-with-padding ajc-text-black ajc-font-default"
      >
        This is a test.
      </div>
    </div>
  </div>
</main>
`;

exports[`PageContent Transferred props and rendering - Component with yellow background verifies the snapshot for the component 1`] = `
<main
  aria-label="Test Title"
  className="ajc-flex-row ajc-flex-justify-center ajc-page-content-panel ajc-spacing-top-small ajc-spacing-bottom-small ajc-width-4k ajc-background-yellow"
  id="main-content"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-left ajc-width-hd"
    >
      <div
        className="ajc-flex-column ajc-width-hd-with-padding ajc-text-black ajc-font-default"
      >
        This is a test.
      </div>
    </div>
  </div>
</main>
`;
